### PR TITLE
Feature/nex 2494 airdrop update contracts according to the discussion

### PR DESCRIPTION
**Pull request by prevostc**
**JIRA task url** (if applicable)

**(If Bug) What was the problem being solved?**

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

**How to reproduce fix / test this PR?**

### CODE REVIEW

File: v2/airdrop-platform/contracts/contracts/MerkleDistributor.sol [hunk 1]
```diff
@@ -45,13 +45,13 @@ contract MerkleDistributor is IMerkleDistributor, TxAuthDataVerifier {
...
     function claim(
         uint256 index,
-        address account,
         uint256 amount,
         bytes32[] calldata merkleProof
     ) public virtual override requireTxDataAuth {
         if (isClaimed(index)) revert AlreadyClaimed();
 
         // Verify the merkle proof.
+        address account = msg.sender;
```
Comment (joelamouche) [inline v2/airdrop-platform/contracts/contracts/MerkleDistributor.sol:54]: You could revert the change on this contract, since it was audited, and add msg.sender in v2/airdrop-platform/contracts/contracts/MerkleDistributorWithDeadline.sol l37 instead

File: v2/airdrop-platform/contracts/contracts/MerkleDistributorWithDeadline.sol [hunk 2]
```diff
@@ -24,13 +25,34 @@ contract MerkleDistributorWithDeadline is MerkleDistributor, Ownable {
...
     function withdraw() external onlyOwner {
         if (block.timestamp < endTime) revert NoWithdrawDuringClaim();
         IERC20(token).safeTransfer(msg.sender, IERC20(token).balanceOf(address(this)));
     }
+
+    /**
+     * @dev Rescues random tokens stuck in the contract.
+     * @param token_ address of the token to rescue.
+     */
+    function rescueToken(address token_) external onlyOwner {
```
Comment (joelamouche) [inline v2/airdrop-platform/contracts/contracts/MerkleDistributorWithDeadline.sol:53]: no 100% sure about this terminology. Have you seen it in other contracts? Maybe retrieve/retrieval

**Review by joelamouche — COMMENTED** [review]
a couple comments

File: v2/airdrop-platform/contracts/contracts/MerkleDistributorWithDeadline.sol (hunk 2 above)
Comment (prevostc) [reply v2/airdrop-platform/contracts/contracts/MerkleDistributorWithDeadline.sol:53]: it's a rescue because it's supposed to be used only if someone sends the wrong token to this contract.
I've seen it at beefy named `inCaseTokenGetStuck` which is arguably better

File: v2/airdrop-platform/contracts/contracts/MerkleDistributor.sol (hunk 1 above)
Comment (prevostc) [reply v2/airdrop-platform/contracts/contracts/MerkleDistributor.sol:54]: MerkleDistributorWithDeadline was also audited afaik since it's part of the original uniswap repo.
Also I think it makes it simpler to have both distributors abide by the same rules and interface.

Comment (joelamouche) [reply v2/airdrop-platform/contracts/contracts/MerkleDistributor.sol:54]: okay yes no nvm

**Review by joelamouche — APPROVED** [review]
LGTM
